### PR TITLE
Limit number of tests in CI comment

### PR DESCRIPTION
**Pull request by miagilepner**
New comment format: 
image

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -101,9 +101,9 @@ jobs:
       - verify-changes
     # Don't run this job for docs/ui only PRs
     if: |
-      always() && (needs.verify-changes.result == 'skipped' ||
+      (always() && needs.verify-changes.result == 'skipped') ||
```
Comment (miagilepner) [inline .github/workflows/ci.yml:104]: this change is needed to correctly cancel go tests when a new commit is pushed to a branch

**Review by mpalmi — APPROVED** [review]
Excellent!

**Review by kubawi — APPROVED** [review]
LGTM!

File: .github/scripts/report_failed_tests.sh [hunk 2]
```diff
@@ -1,22 +1,41 @@
...
-  exit 1
-fi
+  BODY="CI Results:
+All Go tests succeeded! :white_check_mark:"
+else
+  # Remove any rows that don't have a test name
+  # Only keep the test type, test package, test name, and logs column
+  # Remove the scroll emoji
+  # Remove "github.com/hashicorp/vault" from the package name
+  TABLE_DATA=$(echo "$TABLE_DATA" | awk -F\| '{if ($4 != " - ") { print "|" $2 "|" $3 "|" $4 "|" $7 }}' | sed -r 's/ :scroll://' | sed -r 's/github.com\/hashicorp\/vault\///')
```
[the remark this answers is not in the document]
Comment (miagilepner) [reply .github/scripts/report_failed_tests.sh:19]: 🤷‍♀️ I kinda like the full view and I wish that we could use it for comments, but it's impossible to fit it into the comment width. I think it's fine to use different formats

**Pull request merged**